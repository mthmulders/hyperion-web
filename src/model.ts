export interface ApplicationInfo {
    appVersion: string;
    database: string;
    freeMem: string;
    javaVersion: string;
    os: string;
    scalaVersion: string;
    totalMem: string;
};

export interface LiveReading {
    elecCon: number;
    elecProd: number;
    gas: number;
    tariff: string;
    ts: string;
};

export interface Reading {
    electricityLow: number;
    electricityNormal: number;
    gas: number;
    recordDate: Date;
};

export interface RecentReading {
    elecCon: number;
    elecConsNormal: number;
    elecConsLow: number;
    elecProd: number;
    elecProdNormal: number;
    elecProdLow: number;
    gas: number;
    ts: string;
};
