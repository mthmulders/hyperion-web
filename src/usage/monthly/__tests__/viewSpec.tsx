import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import {
    Nav,
    NavItem,
    NavLink,
    TabContent,
    TabPane,
} from 'reactstrap';

import MonthlyUsageView from '../view';
import { UsageRecord } from '../../../model';



describe('<MonthlyUsageView />', () => {
    it('should initially render as a table', () => {
        // Arrange
        const data: UsageRecord[] = [];

        // Act
        const view = shallow(<MonthlyUsageView data={ data } />);

        // Assert
        expect(view.find(TabContent).props().activeTab).toBe('table');
    });

    describe('with the navigation tab clicked', () => {
        it('should render as a graph', (done) => {
            // Arrange
            const data: UsageRecord[] = [];
    
            // Act
            const view = shallow(<MonthlyUsageView data={ data } />);
            view.find(NavLink)
                .findWhere(c => c.props().children === 'Graph')
                .simulate('click');
            
            setTimeout(() => {
                // Assert
                expect(view.find(TabContent).props().activeTab).toBe('graph');
                done();
            }, 500);
        });
    });
});