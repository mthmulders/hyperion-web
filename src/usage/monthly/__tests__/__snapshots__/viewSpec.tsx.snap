// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableView /> should present some data 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table table-borderless"
  >
    <thead>
      <tr>
        <th>
          Date
        </th>
        <th>
          Electricity (low)
        </th>
        <th>
          Electricity (normal)
        </th>
        <th>
          Gas
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          05 Jan 2018
        </td>
        <td>
          2150.714
           kWh
        </td>
        <td>
          2702.641
           kWh
        </td>
        <td>
          3781.676
           m
          <sup>
            3
          </sup>
        </td>
      </tr>
      <tr>
        <td>
          06 Jan 2018
        </td>
        <td>
          2150.714
           kWh
        </td>
        <td>
          2702.641
           kWh
        </td>
        <td>
          3781.676
           m
          <sup>
            3
          </sup>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
